### ISSUE TITLE
React error only in production after upgrading to 3.2.1

### ISSUE THREAD

**Issue by codebycliff** [rails/webpacker#1240]
After we upgraded webpacker from 3.2.0 to 3.2.1, we started getting an error in production. This error only appears in production and works fine while in development using the webpack dev server. I was able to re-produce it locally by running `rails assets:precompile` and commenting out the webpack dev server in our Procfile. The error we are seeing is:

![image](https://user-images.githubusercontent.com/651946/35686524-113227f4-0732-11e8-8d9c-ec241d45abf7.png)

A cursory google search led me to believe this was a problem with how we were importing our components (i.e. named vs default imports). However, we are only using default imports with one component per file, so I don't believe there is a problem there. Simply downgrading to 3.2.0 made the issue go away. Please let me know if I can provide any more information or if there is something I am missing.

**Comment by gauravtiwari** [comment]
@codebycliff Is it possible you are upgrading other packages as well while upgrading webpacker that might be breaking this? How are you doing the upgrade? `yarn upgrade`?

**Comment by codebycliff** [comment]
We use [dependabot](https://dependabot.com/) for dependency upgrades. This bot opens up a pull request for each dependency individually so we can pick and choose what dependencies we want to upgrade. Pinning webpacker and @rails/webpacker to 3.2.0 resolves the problem. However, it's worth noting that the rest of our React components are working fine. The only thing this component does differently than the others is import styles. Here is the problem component:

## BiometricLayout.jsx
```jsx
import React from 'react';
import PropTypes from 'prop-types';

import BiometricsCard from '../components/BiometricsCard';
import BiometricTable from '../components/BiometricTable';

export default function BiometricsLayout(props) {
  const featuredBiometric = props.biometrics[0];
  return (
    <div className="row">
      <div className="col-sm-10 col-center">
        <div className="page-header">
          <h1>Biometrics</h1>
        </div>
      </div>
      <div className="col-sm-10 col-center">
        <p className="margin-bottom-sm">
          <small className="text-muted">
            Last Screening on {featuredBiometric.activity_date} <i className="fa fa-sync" />
          </small>
        </p>
        <div className="row margin-bottom">
          {featuredBiometric.metrics.map(metric => (
            <div key={metric.title} className="col-md-4 col-sm-6">
              <BiometricsCard {...metric} />
            </div>
          ))}
        </div>
      </div>
      {props.biometrics.length > 1 &&
        <div className="col-sm-12">
          <BiometricTable biometrics={props.biometrics} />
        </div>
      }
    </div>
  );
}
```

The actual component causing the problem is the `BiometricCard`. If I remove the rendering of this component, the rest of the `BiometricLayout` component renders fine. The `BiometricCard` component looks like:

## BiometricCard.jsx
```jsx
import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import './BiometricsCard.scss';
import BiometricsCardWrapper from './BiometricsCardWrapper';
import BiometricDescription from './BiometricDescription';

export default class BiometricsCard extends React.Component {
  state = { flipped: false }

  flipCard = () => {
    this.setState(prevState => ({ flipped: !prevState.flipped }));
  }

  hasDescription() {
    return !!this.props.description;
  }

  renderHeader() {
    if (this.hasDescription()) {
      return (
        <div className="bm-card__header">
          <strong
            role="button"
            tabIndex="0"
            aria-pressed={this.state.flipped}
            onClick={this.flipCard}
            className="bm-card__flip_trigger text-black">
            More <i className="fa fa-info-circle" />
          </strong>
        </div>
      );
    }
  }

  renderBackOfCard() {
    if (this.hasDescription()) {
      return (
        <div className="bm-card__back">
          <div className="bm-card__header">
            <strong
              role="button"
              tabIndex="0"
              aria-pressed={this.state.flipped}
              onClick={this.flipCard}
              className="bm-card__flip_trigger text-black">
              Back
            </strong>
          </div>
          <div className="bm-card__body">
            <BiometricDescription
              description={this.props.description}
              desirableRange={this.props.desirable_range} />
          </div>
        </div>
      );
    }
  }

  render() {
    return (
      <BiometricsCardWrapper risk={this.props.risk} flipped={this.state.flipped} description={this.props.description}>
        <div className="bm-card__front">
          {this.renderHeader()}
          <div className="bm-card__body">
            <div className="bm-card__media" style={{ backgroundImage: `url(${this.props.icon})` }} />
            {this.props.stats.map(key => (
              <div key={`${key.value}-${key.label}`} className={`bm-card__measurement bm-card--${key.risk}`}>
                <strong className={classNames('bm-card__total', { 'bm-card__total--big': !key.label })}>{key.value}</strong>
                {key.label && <strong className="bm-card__label">{key.label}</strong>}
              </div>
            ))}
            {this.props.unit && <span className="bm-card__unit">({this.props.unit})</span>}
            <div className="bm-card__title u-sans-serif">{this.props.title}</div>
            <div className="bm-card__status u-sans-serif u-f7">
              {this.props.risk &&
                <span className="bm-card__flip_trigger">
                  <strong
                    role="button"
                    tabIndex="0"
                    aria-pressed={this.state.flipped}
                    onClick={this.flipCard}
                    className="u-underline">
                    {this.props.risk} risk
                  </strong>
                </span>
              }
            </div>
          </div>
        </div>
        {this.renderBackOfCard()}
      </BiometricsCardWrapper>
    );
  }
}
```

The `BiometricDescription` and `BiometricsCardWrapper` are just simple pure components that are exported as the default export. I can post those as well if it helps.

**Comment by shepmaster** [comment]
Possible duplicate of #1235 — could you try the workaround posted at https://github.com/rails/webpacker/issues/1235#issuecomment-362417050 ?

**Comment by IgorDmitriev** [comment]
I think it’s the same as this issue https://github.com/rails/webpacker/issues/1225

**Comment by gauravtiwari** [comment]
@codebycliff Were you able to resolve the issue? Published 3.3.0 today
